apis/telemetry/v1alpha1/telemetry_types.go

```diff
@@ -0,0 +1,38 @@
+// Types in this package are instantiated in the other types in k8ssandra-operator, especially Stargate types and Cassandra types.
+//+kubebuilder:object:generate=true
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type TelemetrySpec struct {
+	Prometheus *PrometheusTelemetrySpec `json:"prometheus,omitempty"`
+}
+type PrometheusTelemetrySpec struct {
+	// Enable the creation of Prometheus serviceMonitors for this resource (Cassandra or Stargate).
+	Enabled *bool `json:"enabled,omitempty"` // A bool flag required here to disambiguate when e.g. the cluster should have telemetry turned on but one DC should have it explicitly turned off.
+	// CommonLabels are applied to all serviceMonitors created.
+	// +optional
+	CommonLabels map[string]string `json:"commonLabels,omitempty"`
+}
+
+type TelemetryStatus struct {
```

Comment:
From my comment above 

> The reason I think status may be important is that any telemetry we integrate with runs the risk that the dependancies (e.g. Prometheus operator) are not installed. In my view, things are very likely to go wrong at this edge of the system, because we have more external dependancy and less control over the things we are integrating with.
> 
> The logical mitigations are validating webhooks and clear status/problem reporting to the user.
> 
> Those considerations notwithstanding, I am very open to removing all status-related code out of this PR, at the very least. Setting status is actually quite complex (not least due to the problems @adutra was raising where you have to change the way that e.g. Stargate and Reaper set status whenever you add an additional status field).
> 
> Let me know which way you want to go.
> 
>